src/crimson/os/seastore/cache.cc

```diff
@@ -68,18 +68,23 @@ void Cache::mark_dirty(CachedExtentRef ref)
     return;
   }
 
-  assert(ref->is_valid());
-  assert(!ref->primary_ref_list_hook.is_linked());
-  intrusive_ptr_add_ref(&*ref);
-  dirty.push_back(*ref);
+  add_to_dirty(ref);
   ref->state = CachedExtent::extent_state_t::DIRTY;
 
   logger().debug("mark_dirty: {}", *ref);
 }
 
-void Cache::retire_extent(CachedExtentRef ref)
+void Cache::add_to_dirty(CachedExtentRef ref)
 {
-  logger().debug("retire_extent: {}", *ref);
+  assert(ref->is_valid());
+  assert(!ref->primary_ref_list_hook.is_linked());
+  intrusive_ptr_add_ref(&*ref);
+  dirty.push_front(*ref);
```

Comment:
@athanatos i assume that you want preserve the order in `dirty`. but i cannot find any clue in the code or comment that explain this.  in `Cache::add_extent()` , we simply append an extent as the end of the list, while in `Cache::add_to_dirty()`, the extent is inserted at the beginning of the list. is this (preservation of the order) a misunderstanding? could you expand on this a little bit?

Reply:
The order we are preserving is the order in which they initially became dirty -- CachedExtent::dirty_from.  This method simply adds to dirty, but it's two callers mark_dirty and replace_extent check whether the extent is already dirty.  In the case of replace_extent, it inserts the extent to the location where prev had already been thus preserving the order.  Otherwise, placing it at the end of the list is the correct choice as any other dirty extents must have been dirtied by a previous journal entry.

Reply:
Looks like I used push_front in add_dirty and rbegin to traverse dirty in get_next_dirty_extents -- I'll swap that as it's not exactly obvious.